src/cryptography/hazmat/primitives/serialization/pkcs7.py

```diff
@@ -221,6 +234,19 @@ def add_recipient(
                 *self._recipients,
                 certificate,
             ],
+            _algorithm=self._algorithm,
+        )
+
+    def set_algorithm(
+        self, algorithm: type[CipherAlgorithm]
+    ) -> PKCS7EnvelopeBuilder:
```

Comment:
Generally we make it an error to call something multiple times.

Reply:
Done!